### ISSUE TITLE
Building gcc (gfortran) on macOS Monterey

### ISSUE THREAD

**Issue by climbfuji** [LLNL/spack#28112]
### Summary

I am not sure if this is a bug or if I am asking for a feature that hasn't been added yet. I am trying to install/build gcc (need gfortran) on macOS Monterey. I have tried the following versions so far in a clean stack environment:
```
spack install gcc@11.2.0 %apple-clang@13.0.0
spack install gcc@11.1.0 %apple-clang@13.0.0
spack install gcc@10.3.0 %apple-clang@13.0.0
spack install gcc@10.3.0
```
The error I get is:
```
...
==> Installing gcc-10.3.0-sn2sgyslqppwjeaidzxvbmzfvsmszhsg
==> No binary for gcc-10.3.0-sn2sgyslqppwjeaidzxvbmzfvsmszhsg found: installing from source
==> Using cached archive: /Users/dom/scratch/spack-test-20211220/spack/var/spack/cache/_source-cache/archive/64/64f404c1a650f27fc33da242e1f2df54952e3963a49e06e73f6940f3223ac344.tar.xz
==> Using cached archive: /Users/dom/scratch/spack-test-20211220/spack/var/spack/cache/_source-cache/archive/ea/eaa00c91e08a5e767f023911a49bc1b2d1a3eea38703b745ab260f90e8da41aa
==> Using cached archive: /Users/dom/scratch/spack-test-20211220/spack/var/spack/cache/_source-cache/archive/28/28c5ab3b564d83dd7e6e35b9c683141a4cb57ee886c5367e54a0828538b3c789
==> Applied patch /Users/dom/scratch/spack-test-20211220/spack/var/spack/repos/builtin/packages/gcc/darwin/gcc-7.1.0-headerpad.patch
==> Applied patch https://gcc.gnu.org/git/?p=gcc.git;a=patch;h=745dae5923aba02982563481d75a21595df22ff8
==> Applied patch /Users/dom/scratch/spack-test-20211220/spack/var/spack/repos/builtin/packages/gcc/zstd.patch
==> Applied patch https://gcc.gnu.org/git/?p=gcc.git;a=patch;h=fc930b3010bd0de899a3da3209eab20664ddb703
==> Ran patch() for gcc
==> gcc: Executing phase: 'autoreconf'
==> gcc: Executing phase: 'configure'
==> gcc: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j4' 'V=1'

4 errors found in build log:
     3753    checking whether times is declared... yes
     3754    checking whether sigaltstack is declared... yes
     3755    checking for struct tms... yes
     3756    checking for clock_t... yes
     3757    checking for F_SETLKW... yes
     3758    checking if mkdir takes one argument... no
  >> 3759    Error: configuring for an unreleased macOS version x86_64-apple-darwin21.1.0
  >> 3760    make[2]: *** [configure-stage1-gcc] Error 1
  >> 3761    make[1]: *** [stage1-bubble] Error 2
  >> 3762    make: *** [all] Error 2
```
Attached is the `spack-build-out.txt` file from the process.
[spack-build-out.txt](https://github.com/spack/spack/files/7756347/spack-build-out.txt)

### Rationale

I am not sure if this is a bug or if I am asking for a feature that hasn't been added yet, therefore logging it as a feature request and not as a bug.

### Description

Installing gcc/gfortran via spack is one of the basic requirements for software stacks. While it is possible to install gcc/gfortran with homebrew and then do the rest with spack, a clean solution using only spack would be preferred.

### Additional information

```
[jedi]  dom@MacBook-Air:~/scratch/spack-test-20211220 [spack]> spack --version
0.17.0-606-494f0053ae
[jedi]  dom@MacBook-Air:~/scratch/spack-test-20211220 [spack]> uname -a
Darwin MacBook-Air.local 21.1.0 Darwin Kernel Version 21.1.0: Wed Oct 13 17:33:23 PDT 2021; root:xnu-8019.41.5~1/RELEASE_X86_64 x86_64
```

### General information

- [X] I have run `spack --version` and reported the version of Spack
- [X] I have searched the issues of this repo and believe this is not a duplicate

**Comment by svigerske** [comment]
Ran into the same and this patch seems to help me to make progress:
https://github.com/gcc-mirror/gcc/commit/11b967577483e51f97d540e9c2c9d1ea76da8122

**Comment by alalazo** [comment]
@climbfuji It seems to be a build error due to GCC. I added the issue to v0.18.0 so we'll be sure to keep track of it.

**Comment by eschnett** [comment]
See https://github.com/spack/spack/pull/27133

**Comment by sethrj** [comment]
Closed by #28662
